packages/comms/src/services/fileSpray.ts

```diff
@@ -100,6 +100,134 @@ export interface DesprayEx {
+        SourceDirectory?: string;
+        DestLogicalName?: string;
+        DestGroupName?: string;
+        DestDirectory?: string;
+        DestIP?: string;
+        DestFilePath?: string;
+        DestFormat?: number;
+        DestNumParts?: number;
+        DestRecordSize?: number;
+        Replicate?: boolean;
+        Overwrite?: boolean;
+        Compress?: boolean;
+        SourceCsvSeparate?: string;
+        SourceCsvQuote?: string;
+        SourceCsvTerminate?: string;
+        SourceCsvEscape?: string;
+        TimeStarted?: string;
+        TimeStopped?: string;
+        StateMessage?: string;
+        MonitorEventName?: string;
+        MonitorSub?: boolean;
+        MonitorShotLimit?: number;
+        SourceDiffKeyName?: string;
+        DestDiffKeyName?: string;
+        Archived?: boolean;
+        encrypt?: string;
+        decrypt?: string;
+        failIfNoSourceFile?: boolean;
+        recordStructurePresent?: boolean;
+        quotedTerminator?: boolean;
+        preserveCompression?: boolean;
+        expireDays?: number;
+        PreserveFileParts?: boolean;
+        FileAccessCost?: number;
+        KbPerSecAve?: number;
+        KbPerSec?: number;
+    };
+    ClusterOrig: string;
+    JobNameOrig: string;
+    isProtectedOrig: boolean;
+    StateOrig: number;
+}
+
+export interface UpdateRequest {
+    ID?: string;
+    DFUServerName?: string;
+    ClusterName?: string;
+    JobName?: string;
+    Queue?: string;
+    User?: string;
+    isProtected?: boolean;
+    Command?: number;
+    CommandMessage?: string;
+    PercentDone?: number;
+    SecsLeft?: number;
+    ProgressMessage?: string;
+    SummaryMessage?: string;
+    State?: number;
+    SourceLogicalName?: string;
+    SourceIP?: string;
+    SourceFilePath?: string;
+    SourceDali?: string;
+    SourceRecordSize?: number;
+    SourceFormat?: number;
+    RowTag?: string;
+    SourceNumParts?: number;
+    SourceDirectory?: string;
+    DestLogicalName?: string;
+    DestGroupName?: string;
+    DestDirectory?: string;
+    DestIP?: string;
+    DestFilePath?: string;
+    DestFormat?: number;
+    DestNumParts?: number;
+    DestRecordSize?: number;
+    Replicate?: boolean;
+    Overwrite?: boolean;
+    Compress?: boolean;
+    SourceCsvSeparate?: string;
+    SourceCsvQuote?: string;
+    SourceCsvTerminate?: string;
+    SourceCsvEscape?: string;
+    TimeStarted?: string;
+    TimeStopped?: string;
+    StateMessage?: string;
+    MonitorEventName?: string;
+    MonitorSub?: boolean;
+    MonitorShotLimit?: number;
+    SourceDiffKeyName?: string;
+    DestDiffKeyName?: string;
+    Archived?: boolean;
+    encrypt?: string;
+    decrypt?: string;
+    failIfNoSourceFile?: boolean;
+    recordStructurePresent?: boolean;
+    quotedTerminator?: boolean;
+    preserveCompression?: boolean;
+    expireDays?: number;
+    PreserveFileParts?: boolean;
+    FileAccessCost?: number;
+    KbPerSecAve?: number;
+    KbPerSec?: number;
+}
```

Comment:
These are duplicated from the generated code?  You can use `Partial<FileSpray.UpdateDFUWorkunit>` to achieve the same thing?

Reply:
Yeah, I actually started there. But I couldn't work out why type the ECL Watch side was clashing because of that wu: {...} property that contains all the actual properties. I'm not sure what I was missing, but it would either end up submitting something like `...wu.wu.Jobname: "xyz"...` or `...wu: {...}, JobName: "xyz"...` which is invalid and was ignored...

Reply:
Do you think this is an issue with the wsdl generation or with the WSDL definition?

Reply:
I think it's the generation. Seems like it's not taking into account the "minOccurs" attribute to indicate whether the properties in the created interfaces should be optional.